### ISSUE TITLE
MapEvaluator.compute_npred() fails with edisp param and 3D models

### ISSUE THREAD

**Issue by Bultako** [gammapy/gammapy#1956]
When calling MapEvaluator.compute_npred() for a MapEvaluator instantiated with different shapes in the geometries of the SkyModel and the edisp param,  I have the error below.

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-31-3546407d1fbf> in <module>
      1 #dnde = evaluator.compute_dnde()
      2 #flux = evaluator.compute_flux()
----> 3 npred = evaluator.compute_npred()
      4 npred_map = WcsNDMap(geom3d, npred)

~/git/gammapy/gammapy/gammapy/cube/fit.py in compute_npred(self)
    247             npred = self.apply_psf(npred)
    248         if self.edisp is not None:
--> 249             npred = self.apply_edisp(npred)
    250         if self.background:
    251             npred.data += self.background.data

~/git/gammapy/gammapy/gammapy/cube/fit.py in apply_edisp(self, npred)
    223         loc = npred.geom.get_axis_index_by_name("energy")
    224         data = np.rollaxis(npred.data, loc, len(npred.data.shape))
--> 225         data = np.dot(data, self.edisp.pdf_matrix)
    226         data = np.rollaxis(data, -1, loc)
    227         e_reco_axis = MapAxis.from_edges(

ValueError: shapes (73,73,99) and (500,500) not aligned: 99 (dim 2) != 500 (dim 0)
```

**Comment by cdeil** [comment]
@Bultako - Can you try to make a minimal example to trigger the issue?

Note that if `psf` or `background` isn't involved in the issue, you should be able to just not pass those to `MapEvaluator`, they are optional.

I think the way [MapEvaluator](https://docs.gammapy.org/0.8/api/gammapy.cube.MapEvaluator.html) should work is to allow any `model`, but for `exposure`, `background`, `psf` and `edisp` they have to have consistent binnings on input.

So is the problem really with the input `model`?

**Comment by Bultako** [comment]
Ok.
I have an example to reproduce the issue.

In `simulate_3d.ipynb` notebook, in `cell [12]` add edisp=edisp when instantiating the `MapEvaluator`, then we have the error when executing `cell [13]`. I think this comes because in internal `apply_disp` method we try to make an impossible matrix multiplication of different number of dimensions.

https://github.com/gammapy/gammapy/blob/d47188fdfcfd1af81aa633380490343690bde290/gammapy/cube/fit.py#L225

**Comment by cdeil** [comment]
I see. The problem is that in `simulate_3d.ipynb` we don't create an EDISP matrix with energy axes that match the maps. This can be fixed by passing appropriate `e_reco` and `e_true` when creating `edisp` in `simulate_3d.ipynb`
```
edisp = irfs["edisp"]..to_energy_dispersion(offset, e_reco=e_reco, e_true=e_true)
```
similar to here:
https://nbviewer.jupyter.org/github/gammapy/gammapy/blob/master/tutorials/analysis_3d.ipynb#Prepare-IRFs

It didn't fail so far because `edisp` isn't used at all.

Also, when updating the notebook, I'd suggest to delete the `iminuit` section at the end, not needed / useful any more.

@Bultako or @AtreyeeS  - Do you have time to send a PR to update `simulate3d.ipynb`?

**Comment by Bultako** [comment]
Ok.
I understand.
I've made a PR updating the notebook as suggested.
Closing this.

**Comment by cdeil** [comment]
Fixed by @Bultako in #1957.
